### DISCUSSION TITLE
Speed - initialization and rendering

### DISCUSSION THREAD

**Discussion by klokan** [maplibre/maplibre#160] [Cross-project feature proposals]
Optimization related to initialization time of the rendering libraries, delay before the map appears in the app.
Rendering performance improvements, measurement, test and best-practice documentation.

**Ideas / tickets towards this direction should be proposed in different CORE projects of MapLibre.**

**Comment by lseelenbinder** [comment]
Probably the first step is building a performance test suite so we can track this by MR (to see regressions and improvements), and then start a long set of epics to address worst-offenders incrementally.

**Reply by rotu** [reply]
@lseelenbinder in what way are the current benchmarks insufficient?

https://github.com/maplibre/maplibre-gl-js/blob/a81b2b1549ba43a18ca3d7746178023c3d335abd/test/bench/README.md

**Reply by lseelenbinder** [reply]
@rotu I was speaking holistically about all projects: establish a baseline, ensure the baseline is met, and then iterate. For example, Native is behind GL JS with automated performance tests across a representative suite of platforms.

**Reply by rotu** [reply]
> I was speaking holistically about all projects: establish a baseline, ensure the baseline is met, and then iterate

Okay. It looks like that's what the current benchmark tooling is aspiring to do. Are performance issues around "initializing and rendering" visible in the current tooling? How does the tooling need to change (versus the tools exist and devs just need to use them more effectively)?

> For example, Native is behind GL JS with automated performance tests across a representative suite of platforms.

I'm sorry I don't understand what you mean here.

**Reply by lseelenbinder** [reply]
>  I'm sorry I don't understand what you mean here.

[https://github.com/maplibre/maplibre-gl-native/]()'s performance tests are behind GL JS, as I understand it. (They exist, but we need to run them in a consistent CI environment so we can track it and act on it.)

> Are performance issues around "initializing and rendering" visible in the current tooling? How does the tooling need to change (versus the tools exist and devs just need to use them more effectively)?

I haven't done the research to answer this in these terms. My point was just that to effectively address performance issues, we have to have a baseline performance test suite that tracks the performance we care about--if it already exists for the important projects, great, no action necessary.

**Comment by wipfli** [comment]
This project was selected to become a Bounty Direction. You find a tracking issue at https://github.com/maplibre/maplibre/issues/192
